### PR TITLE
Refactor disciplina config interface for new YAML parser

### PR DESCRIPTION
**Pull request by mkaito**
Some things can't be moved, because sections are sometimes parsed as a unit, and
the config itself is more verbose than necessary because many options lack sane
defaults.

### CODE REVIEW

File: deployments/cluster/educator.nix [hunk 1]
```diff
@@ -11,45 +11,70 @@ in
...
+          genNew = true;
+        };
+
+        api = {
+          serverParams.addr = "0.0.0.0:4040";
+          botParams = {
+            operationsDelay = "3s";
+            seed = "super secure"; # I don't know whether or not this is sensitive data
+          };
+          studentAPINoAuth.enabled = false;
```
Comment (yorickvP) [inline deployments/cluster/educator.nix]: does that have auth or not?

File: deployments/cluster/witness.nix [hunk 2]
```diff
@@ -13,44 +13,44 @@ in
...
+      api.addr = "0.0.0.0:4030";
+    };
+
     args = let
       cat = path: ''"$(cat "${path}")"'';
-      stateDir = "/var/lib/disciplina-witness";
-
-      publicIP = ''"$(curl "http://169.254.169.254/latest/meta-data/public-ipv4")"'';
-      privateIP = ''"$(curl "http://169.254.169.254/latest/meta-data/local-ipv4")"'';
+      publicIP = ''"$(curl "http://169.254.169.254/latest/meta-data/public-ipv4" 2>/dev/null)"'';
```
Comment (yorickvP) [inline deployments/cluster/witness.nix]: what could go wrong?

File: deployments/cluster/educator.nix [hunk 3]
```diff
@@ -11,45 +11,70 @@ in
...
+          maxPending = 100;
+        };
+
+        keys = {
+          path = "${witness.appDir}/educator.key";
+          genNew = true;
+        };
+
+        api = {
+          serverParams.addr = "0.0.0.0:4040";
```
Comment (yorickvP) [inline deployments/cluster/educator.nix]: who needed consistency anyways.

File: deployments/cluster/educator.nix (hunk 1 above)
Comment (mkaito) [reply deployments/cluster/educator.nix]: This is one of those options that should just have a sane default option. Enabling this option disables auth. Confusing.

File: deployments/cluster/witness.nix (hunk 2 above)
Comment (mkaito) [reply deployments/cluster/witness.nix]: There're reasons I `set -euxo pipefail` explicitly in the service script. This is one of those reasons.

File: deployments/cluster/educator.nix (hunk 3 above)
Comment (mkaito) [reply deployments/cluster/educator.nix]: ikr

**Pull request merged**